internal/controllers/import_controller.go

```diff
@@ -246,13 +247,21 @@ func (r *CAPIImportReconciler) shouldAutoImport(ctx context.Context, capiCluster
 	return autoImport, nil
 }
 
-func (r *CAPIImportReconciler) reconcileDelete(ctx context.Context, _ *clusterv1.Cluster) (ctrl.Result, error) {
+func (r *CAPIImportReconciler) reconcileDelete(ctx context.Context, capiCluster *clusterv1.Cluster) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 	log.Info("Reconciling CAPI Cluster Delete")
 
-	// TODO: add implementation
+	if hasAnnotation(capiCluster, clusterImportedAnnotation) {
+		if err := r.Client.Delete(ctx, capiCluster); err != nil && !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("error deleting %s cluster with imported annotation", capiCluster.Name)
```

Comment:
seems like `err` value is ignored in the error message.

Reply:
Thanks, added it to error message